instrumentation/apache-httpclient/apache-httpclient-5.0/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v5_0/ResponsePayloadBridge.java

```diff
@@ -0,0 +1,157 @@
+    ResponsePayloadBridge bridge = context.get(CONTEXT_KEY);
+    if (bridge != null && !bridge.payloadRetrieved) {
+      bridge.payloadRetrieved = true;
+
+      // If chunked response, ensure we strip the chunk header/trailer from the payload
+      if (bridge.isChunked) {
+        // Capture chunk header indicating size of chunk
+        byte[] chunkHeaderBytes = new byte[15];
+        int payloadStart = -1;
+        int chunkSize = 0;
+        for (int i = 0; i < bridge.payloadBuffer.length; i++) {
+          if ((char)bridge.payloadBuffer[i] == '\r') {
+            if ((char)bridge.payloadBuffer[i+1] == '\n') {
+              // Found CRLF at end of chunk header
+              final String chunkHeaderString = new String(chunkHeaderBytes, 0, i, StandardCharsets.UTF_8);
+              payloadStart = i+2;
+              // Convert the Chunk header from HEX into the number of bytes the chunk contains
+              chunkSize = Integer.parseInt(chunkHeaderString, 16);
+              break;
+            }
+          } else {
+            // Haven't found the end of the chunk header, capture the byte we found
+            chunkHeaderBytes[i] = bridge.payloadBuffer[i];
+          }
+        }
+
+        if (payloadStart > -1) {
+          if (bridge.payloadBufferLength < chunkSize) {
+            // We haven't appended enough data to return anything
+            return null;
+          }
+
+          if (bridge.isGzipped) {
+            // Return the payload chunk only, encoded as Base64
+            return Base64.getEncoder().encodeToString(
+                new String(bridge.payloadBuffer, payloadStart, chunkSize,
+                    StandardCharsets.UTF_8).getBytes(
+                    StandardCharsets.UTF_8));
+          } else {
+            // Return the payload chunk only
+            return new String(bridge.payloadBuffer, payloadStart, chunkSize,
+                StandardCharsets.UTF_8);
+          }
+        }
+      }
+
+      return new String(bridge.payloadBuffer, 0, bridge.payloadBufferLength, StandardCharsets.UTF_8);
+    }
+
+    return null;
+  }
+
+  public static void responseHeaders(Context context, HttpMessage httpMessage) {
+    ResponsePayloadBridge bridge = context.get(CONTEXT_KEY);
+    if (bridge != null && httpMessage.getHeaders() != null) {
+      bridge.headers = httpMessage.getHeaders();
+      checkForGzip(bridge);
+      checkForChunked(bridge);
+    }
+  }
+
+  public static boolean isGzipped(Context context) {
+    ResponsePayloadBridge bridge = context.get(CONTEXT_KEY);
+    if (bridge != null) {
+      return bridge.isGzipped;
+    }
+
+    return false;
+  }
+
+  private static void checkForGzip(ResponsePayloadBridge bridge) {
+    Optional<Header> contentEncodingHeader = Arrays.stream(bridge.headers)
+        .filter(header -> header.getName().equals("Content-Encoding"))
+        .findFirst();
+    if (contentEncodingHeader.isPresent() && contentEncodingHeader.get().getValue().equals("gzip")) {
```

Comment:
I think it is better to use `equalsIgnoreCase` using `"gzip"` as first member, i.e.:
```
"gzip".equalsIgnoreCase(contentEncodingHeader.get().getValue())
```